### DISCUSSION TITLE
ANCOMBC2 code

### DISCUSSION THREAD

**Discussion by Lobna-H** [FrederickHuangLin/ANCOMBC#229] [Q&A]
I am trying to perform differential abundance for microbial community composition by ANCOMBC2 using R.
I have 3 samples from 4 dependent groups (group 1, group 2, group 3, group 4) each group contains 3 samples. group1 is the beginning point (timepoint0), group 2 is group 1 after treatment 1 for a period of time (timepoint1), group 3 is group 2 after treatment 2 for a period of time (timepoint3), group 4 is group 3 after treatment 3 for a period of time (timepoint4). group 1 is the reference group.

Here my script

output <- ancombc2(data = pseq5, assay_name = "counts", tax_level = "Phylum",
                  fix_formula = " groups + Timepoint",
                  rand_formula = "(Timepoint | subject)",
                  p_adj_method = "holm",
                  prv_cut = 0.10, lib_cut = 1000, s0_perc = 0.05,
                  group = "groups", struc_zero = TRUE, neg_lb = TRUE,
                  alpha = 0.05, n_cl = 3, verbose = TRUE,
                  global = TRUE, pairwise = TRUE, dunnet = TRUE, trend = TRUE,
                  iter_control = list(tol = 1e-2, max_iter = 20, 
                                      verbose = TRUE),
                  em_control = list(tol = 1e-5, max_iter = 100),
                  lme_control = lme4::lmerControl(),
                  mdfdr_control = list(fwer_ctrl_method = "holm", B = 100),
                  trend_control = list(contrast = list(matrix(c(1, 0, 0,
                                                               -1, 1, 0,
                                                               0, -1, 1),
                                                              nrow = 3, 
                                                              byrow = TRUE)),
                                       node = list(3),
                                       solver = "ECOS",
                                       B = 100))

I got the error 

Obtaining initial estimates ...
Error: Estimation failed for the following covariates:
Timepointt1, Timepointt2, Timepointt3
Please ensure that these covariates do not have missing values and check for multicollinearity before re-estimating the model
In addition: Warning message:
Small sample size detected for the following group(s): 
group1, group2, group3, group4
Variance estimation would be unstable when the sample size is < 5 per group    

Could you please help solve this error

**Comment by FrederickHuangLin** [comment]
Hello @Lobna-H,

If I've grasped your code correctly, it appears that your `timepoint` and `group` variables contain identical information, potentially leading to a multicollinearity problem. To investigate whether this issue exists, you can perform a simple linear regression `lm` using the formula `groups + Timepoint` with your microbial abundance data.

Best regards,
Huang

**Reply by simone-anza** [reply]
Hi @FrederickHuangLin 

thank you for the reply, I am not sure about this approach and I am having the same issue of understanding which variable is multicollinear for my ancombc model for gut microbiome analyses. Let's say I want to check for multicollinearity of varA and varB, what is my response variable in my `lm` considering that microbial abundance data is compositional? Thank you for this tool! 

Best,
Simone
